### PR TITLE
[ADD] estate: modify user model view and behavior on properties and offer creation/deletion T#70778

### PR DESCRIPTION
**Pull request by Gilmaciasr**
In this update the following has been applied:
-Inherited user view to show just few fields and add a tab to real estate properties, now you can see which properties are being sold by each salesman(user) directly on user view.
-Apply some methods to prevent deletion of properties, just new or canceled can be deleted. A user error is prompt if so.
-On offer creation a restriction has been applied, new offers should be higher than the previous one in order to be saved.
Link to the lesson:
https://www.odoo.com/documentation/16.0/developer/tutorials/getting_started/13_inheritance.html
Links to the videos:
https://youtu.be/RimhVknYQ0w
https://youtu.be/2quJ-C075to

### CODE REVIEW

**Comment by Gilmaciasr** [comment]
@JesusValdez96 ready for the show!

File: estate/models/estate_property.py [hunk 1]
```diff
@@ -138,3 +144,10 @@ def cancel_button_action(self):
                 raise exceptions.UserError(_("Sold properties cannot be canceled"))
             record.states = "canceled"
         return True
+
+    def check_offer(self, new_offer_price):
+        for record in self:
+            prices = record.mapped("offer_ids.price")
+            if prices and new_offer_price < max(prices):
+                raise exceptions.UserError(_("The new offer should be higher than %s", max(prices)))
+            return True
```
Comment (JesusValdez96) [inline estate/models/estate_property.py:153]:
```suggestion
        for record in self:
            prices = record.mapped("offer_ids.price")
            if prices and new_offer_price < max(prices):
                raise exceptions.UserError(_("The new offer should be higher than %s", max(prices)))
        return True
```

File: estate/views/res_users_views.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<odoo>
+
+  <record id="res_users_view_form" model="ir.ui.view">
+      <field name="name">res.users.form.inherit</field>
+      <field name="model">res.users</field>
+      <field name="inherit_id" ref="base.view_users_form" />
+      <field name="priority">110</field>
+      <field name="arch" type="xml">
+      <!-- IN THIS CASE JUST FEW FIELDS ARE NECESSARY TO SHOW -->
+      <xpath expr="//form" position="replace">
```
Comment (JesusValdez96) [inline estate/views/res_users_views.xml:11]: don't use position="replace" if not necessary

Comment (JesusValdez96) [inline estate/views/res_users_views.xml:11]: why did you overwrite the whole form? it's better only add the new page to the notebook section with a specific expr to preferences page and position="after"

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]

File: estate/views/res_users_views.xml (hunk 2 above)
Comment (Gilmaciasr) [reply estate/views/res_users_views.xml]: @JesusValdez96 I misunderstood the instructions, i noticed on later exercises and there i changed it, but i'll fix it at this review right away,

**Pull request merged**